dandiapi/api/views/dashboard.py

```diff
@@ -82,6 +83,39 @@ def _users(self):
         )
 
 
+def mailchimp_csv_view(request: HttpRequest) -> StreamingHttpResponse:
+    """Generate a Mailchimp-compatible CSV file of all active users."""
+    # In production, there's a placeholder user with a blank email that we want
+    # to avoid.
+    users = User.objects.filter(is_active=True).exclude(email='')
+    data = users.values_list('email', 'first_name', 'last_name').iterator()
+
+    def streaming_output():
```

Comment:
```suggestion
    def streaming_output() -> Iterator[tuple[str, str, str]]:
```

Reply:
IIUC, this generator yields strings. The reason we're using that `Echo` pseudo-buffer thing is to cause the  `writeheader()` and `writerow()` methods to return the string that they write. (Apparently, those methods just return the result of calling `.write()` on the underlying filelike object. By contrast, if we did this with a real filelike, whose `.write()` method returns the number of bytes written, we'd end up sending those byte counts to the streaming response.) It's sort of a mess 🫤 but it works.

If my analysis is correct, I'll update the suggestion to use `Iterator[str]` and commit it. Let me know what you think.

Reply:
Ah you're right, I verified it locally and it is indeed a `Iterator[str]`.